File: cousins in binary tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    void dfs(TreeNode root,int dep[],HashMap<Integer,Integer>par,int d){
        if(root==null){return ;}
        dep[root.val]=d;
        if(root.left!=null){
            par.put(root.left.val,root.val);
            dfs(root.left,dep,par,d+1);
        }
        if(root.right!=null){
            par.put(root.right.val,root.val);
            dfs(root.right,dep,par,d+1);
        }
        return ;
    }
    public boolean isCousins(TreeNode root, int x, int y) {
        int dep[]=new int[101];
        dep[root.val]=0;
        HashMap<Integer,Integer>par=new HashMap<Integer,Integer>();
        par.put(root.val,root.val);
        dfs(root,dep,par,0);
        return (dep[x]==dep[y])&&(par.get(x)!=par.get(y));
    }
}

File: cousins in binary tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
void dfs(TreeNode* root,unordered_map<int,int>&par,int dep[],int d,vector<int>&v){
    if(root==nullptr){return ;}
    dep[root->val]=d;
    v.emplace_back(root->val);
    if(root->left){
        par[root->left->val]=root->val;
        dfs(root->left,par,dep,d+1,v);
    }
    if(root->right){
        par[root->right->val]=root->val;
        dfs(root->right,par,dep,d+1,v);
    }
}
    bool isCousins(TreeNode* root, int x, int y) {
        unordered_map<int,int>par;
        vector<int>v;
        int dep[101]={0};
        dep[root->val]=0;
        par[root->val]=root->val;
        dfs(root,par,dep,0,v);
        for(int x:v){
            cout<<x<<" ,dep:"<<dep[x]<<",par:"<<par[x]<<"\n";
        }
        return  par[x]!=par[y]&&(dep[x]==dep[y]);
    }
};